app/scripts/models/auth_brokers/fx-desktop.js

```diff
@@ -65,6 +71,36 @@ define([
           self, options);
     },
 
+    /**
+      * Is used to ask the client about capabilities before rendering the views
+      *
+      * @returns {Promise}
+      */
+    beforeRender: function () {
+      var self = this;
+      var defer = p.defer();
+      var result = null;
+
+      self.request(self._commands.CAPABILITIES)
+        .then(function (response) {
+          if (response && response.data && _.isObject(response.data)) {
+            result = response.data;
+
+            self.setCapabilities(result);
+          }
+
+          defer.resolve();
+        });
+
+      self.window.setTimeout(function () {
```

Comment:
It seems like this logic should be a part of the channel again, possibly making the timeout configurable. We removed timeout's being an error a while back, I forget what the reasoning was.